apps/db/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/db/InntektsmeldingRepository.kt

```diff
@@ -25,25 +28,51 @@ class InntektsmeldingRepository(
     private val logger = logger()
     private val sikkerLogger = sikkerLogger()
 
-    fun hentNyesteEksternEllerInternInntektsmelding(forespoerselId: UUID): Triple<SkjemaInntektsmelding?, Inntektsmelding?, EksternInntektsmelding?> =
+    fun hentNyesteEksternEllerInternInntektsmelding(forespoerselId: UUID): LagretInntektsmelding? =
         Metrics.dbInntektsmelding.recordTime(InntektsmeldingRepository::hentNyesteEksternEllerInternInntektsmelding) {
             transaction(db) {
                 InntektsmeldingEntitet
                     .select(
-                        InntektsmeldingEntitet.skjema,
                         InntektsmeldingEntitet.dokument,
+                        InntektsmeldingEntitet.skjema,
                         InntektsmeldingEntitet.eksternInntektsmelding,
+                        InntektsmeldingEntitet.innsendt,
                     ).where { InntektsmeldingEntitet.forespoerselId eq forespoerselId.toString() }
                     .orderBy(InntektsmeldingEntitet.innsendt, SortOrder.DESC)
                     .limit(1)
                     .map {
-                        Triple(
-                            it[InntektsmeldingEntitet.skjema],
+                        Quadruple(
```

Comment:
Kan man ikke droppe map og bare gjøre 

val inntektsmelding = result[InntektsmeldingEntitet.dokument]
                    val skjema = result[InntektsmeldingEntitet.skjema]
                    val eksternInntektsmelding = result[InntektsmeldingEntitet.eksternInntektsmelding]
                    val mottatt = result[InntektsmeldingEntitet.innsendt]

Synes ikke at Quadruple-objektet hjelper noe særlig på lesbarhet i hvert fall

Reply:
Jeg brukte Quadruple for å holde transaksjonen mot databasen kortest mulig, siden det som kommer etter ikke er en del av spørringen, men det er kanskje ikke så farlig 🤔